### PR TITLE
chore: refactor controllers

### PR DESCRIPTION
**Pull request by im-adithya**
Removes `PermissionsService` and adds `ControllerService`

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -21,31 +21,28 @@ import (
...
 	"github.com/getAlby/nostr-wallet-connect/service/keys"
 	"github.com/getAlby/nostr-wallet-connect/utils"
 )
 
 type api struct {
-	db             *gorm.DB
-	dbSvc          db.DBService
-	cfg            config.Config
-	svc            service.Service
-	permissionsSvc permissions.PermissionsService
```
Comment (im-adithya) [inline api/api.go:35]: We are adding this svc just to make use of the `GetBudgetUsage` function, moved `GetStartOfBudget` to utils folder and after that it is just a one line sql query which I moved to this file directly.

Comment (rolznz) [reply api/api.go:35]: I would not like to duplicate that SQL query, this is the core logic to see if an app has permission to make a payment

Comment (im-adithya) [reply api/api.go:35]: Then we might have to add the controllersSvc just for that function, should I do it?

File: nip47/controllers/controller_tests/multi_pay_invoice_controller_test.go [hunk 2]
```diff
@@ -135,12 +137,13 @@ func TestHandleMultiPayInvoiceEvent_WithPermission(t *testing.T) {
...
-	NewMultiPayInvoiceController(svc.LNClient, svc.DB, svc.EventPublisher).
-		HandleMultiPayInvoiceEvent(ctx, nip47Request, dbRequestEvent.ID, app, checkPermission, publishResponse)
+	controllersSvc := controllers.NewControllersService(svc.DB, svc.EventPublisher, svc.LNClient)
+
+	controllersSvc.HandleMultiPayInvoiceEvent(ctx, nip47Request, dbRequestEvent.ID, app, checkPermission, publishResponse)
 
 	assert.Equal(t, 2, len(responses))
 	for i := 0; i < len(responses); i++ {
-		assert.Equal(t, "123preimage", responses[i].Result.(payResponse).Preimage)
+		assert.Equal(t, "123preimage", responses[i].Result.(controllers.PayResponse).Preimage)
```
Comment (rolznz) [inline nip47/controllers/controller_tests/multi_pay_invoice_controller_test.go:146]: I don't think we should move the tests to a new package, now we have to make models public just for the tests - this is why I put them side by side with the code. A  lot of other projects do this too.

File: nip47/event_handler.go [hunk 3]
```diff
@@ -20,6 +19,10 @@ import (
 )
 
 func (svc *nip47Service) HandleEvent(ctx context.Context, sub *nostr.Subscription, event *nostr.Event, lnClient lnclient.LNClient) {
+	if !svc.controllersService.HasLNClient() {
```
Comment (rolznz) [inline nip47/event_handler.go:22]: this feels like a hack, can we pass the LNClient to the individual controller handler methods?

File: nip47/notifications/nip47_notifier.go [hunk 4]
```diff
@@ -77,7 +77,7 @@ func (notifier *Nip47Notifier) notifySubscribers(ctx context.Context, notificati
 	notifier.db.Find(&apps)
 
 	for _, app := range apps {
-		hasPermission, _, _ := notifier.permissionsSvc.HasPermission(&app, permissions.NOTIFICATIONS_PERMISSION, 0)
+		hasPermission, _, _ := notifier.controllersSvc.HasPermission(&app, controllers.NOTIFICATIONS_PERMISSION, 0)
```
Comment (rolznz) [inline nip47/notifications/nip47_notifier.go:80]: I think this is coupling notifications to controllers - these are two separate things and should not be related

**Comment by rolznz** [comment]
@im-adithya @bumi I feel like we unfortunately went a bit backwards on this PR.

I suggest we make the minimal change to create a common interface for the controller methods and then consider the other stuff as a follow-up. Does that sound ok?

**Comment by rolznz** [comment]
@im-adithya I talked with @bumi  and we thought for now it's not worthwhile to make this change as it's opening up some other issues (as per the comments). For now we'd like to close this one and focus on some more other things (changing the authentication to not use a session cookie for example). I will create a new issue keep track of this though and link to this PR.

**Pull request closed without merging**